tket/src/Transformations/ThreeQubitSquash.cpp

```diff
@@ -108,22 +108,28 @@ class QInteraction {
 typedef std::unique_ptr<QInteraction> iptr;
 
 // Candidate substitution for a 2- or 3-qubit circuit.
-static Circuit candidate_sub(const Circuit &circ) {
+static Circuit candidate_sub(const Circuit &circ, OpType target_2q_gate) {
   unsigned n_qb = circ.n_qubits();
   if (n_qb == 2) {
     Circuit repl = two_qubit_canonical(get_matrix_from_2qb_circ(circ));
-    clifford_simp(false).apply(repl);
+    if (target_2q_gate == OpType::CX) {
+      clifford_simp(false).apply(repl);
+    }
     return repl;
   } else {
     TKET_ASSERT(n_qb == 3);
-    Circuit repl = three_qubit_synthesis(get_3q_unitary(circ));
-    // TODO: for now we decompose all the way to CX. In the future, it's worth
-    // considering keeping TK2, and decomposing to CX (or other gates) later
-    // when necessary.
-    normalise_TK2().apply(repl);
-    decompose_TK2().apply(repl);
-    clifford_simp(false).apply(repl);
-    return repl;
+    if (target_2q_gate == OpType::CX) {
+      Circuit repl = three_qubit_synthesis(get_3q_unitary(circ));
+      normalise_TK2().apply(repl);
+      decompose_TK2().apply(repl);
+      clifford_simp(false).apply(repl);
```

Comment:
If we `squash_1qb_to_tk1` in the `TK2` case, would it not make sense to do it here too? I would also move out `three_qubit_synthesis(get_3q_unitary(circ))` from the if-else clause to highlight only the differences between the two OpTypes, but that's subjective taste.